react/features/base/tracks/reducer.js

```diff
@@ -127,15 +127,7 @@ ReducerRegistry.register('features/base/tracks', (state = [], action) => {
         return state.map(t => track(t, action));
 
     case TRACK_ADDED: {
-        let withoutTrackStub = state;
-
-        if (action.track.local) {
-            withoutTrackStub
-                = state.filter(
-                    t => !t.local || t.mediaType !== action.track.mediaType);
-        }
-
-        return [ ...withoutTrackStub, action.track ];
+        return [ ...state, action.track ];
```

Comment:
why is this no longer needed?

Reply:
In the application, the old tracks are always removed first before adding the new track to the redux - https://github.com/jitsi/jitsi-meet/blob/c5fbe14de2964391b8eb5300b02317aad973a3a8/react/features/base/tracks/actions.js#L322. I don't see when we would run into a case where there would be more than 1 track of the same media type.